docs/learn/encyclopedia/data-format/xdr.mdx

```diff
@@ -38,3 +41,219 @@ XDR is specified in [RFC 4506](http://tools.ietf.org/html/rfc4506.html) and is s
+
+<CodeExample>
+
+```typescript
+import { Address } from "@stellar/stellar-sdk";
+
+// suppose `scAddr` is an instance of `xdr.ScAddress` we're parsing
+const address = Address.fromScAddress(scAddr);
+```
+
+```python
+import Address from stellar_sdk;
+
+// suppose `scAddr` is an instance of `xdr.ScAddress` we're parsing
+addr = Address.from_xdr_sc_address(scAddr)
+```
+
+</CodeExample>
+
+Always try to find a higher-level abstraction before digging into the raw structure.
+
+#### Structures
+
+Like in any C-like language, a structure is a packed object containing a bunch of fields of arbitrary types. Structures can themselves contain structures, so be sure you traverse the entire tree of fields when you're constructing one.
+
+Let's take, for example, the [`ContractEvent`](https://github.com/stellar/stellar-xdr/blob/c1a58e2f81a4d1b02adc6db9914e6f51960e1ce3/Stellar-ledger.x#L339-L358) structure. Its definition is a little complicated, so let's break it down:
+
+- An `ExtensionPoint` is a way for the protocol to be extended while maintaining backwards binary compatibility.
+
+It's basically an empty slot into which you can add variations in the future. The [`SorobanTransactionMeta` structure](https://github.com/stellar/stellar-xdr/blob/c1a58e2f81a4d1b02adc6db9914e6f51960e1ce3/Stellar-ledger.x#L410) is a great example of this in action: When the structure was initially added to the protocol, it just contained the details of an invocation (events and a return value). Later, we needed more details about the invocation in the resulting metadata, so `SorobanTransactionMetaExt` was injected into the `ExtensionPoint` with a [`V1` variant](https://github.com/stellar/stellar-xdr/blob/c1a58e2f81a4d1b02adc6db9914e6f51960e1ce3/Stellar-ledger.x#L368C8-L368C35) containing metadata about fees charged as a result of the invocation.
+
+- The `contractID` should be self-explanatory: it's the hash of the contract ID that caused this event. It's a pointer, though, because not all events are related to a specific contract. Some may come from the system, diagnostics, etc.
+
+It should not be surprising that the field is stored as a `Hash`: even though at the ecosystem layer we refer to contracts by their "strkey" or string representation (in the form `C...`), these are actually "friendlier" representations of a raw SHA-256 hash. This is common at the protocol layer: the simplest, most compact representation will always be used. There are generally SDK tools to make translating these
```

Comment:
this seems to "drift off" without completing a sentence

Reply:
gotta keep your readers on their toes man